spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'PostsController', type: :request do
+  describe 'GET /index' do
+    context 'when the page is loaded' do
+      it 'returns a 200 status code and render template for index' do
+        get '/users/1/posts'
+        expect(response).to have_http_status(200)
+        expect(response).to render_template(:index)
+      end
+      it 'includes pagination' do
+        get '/users/1/posts'
+        expect(response.body).to include('Pagination')
+      end
+    end
+  end
+
+  describe 'GET /show' do
+    context 'when the page is loaded' do
+      it 'returns a 200 status code and render template for show' do
+        get '/users/1/posts', params: { id: 1 }
+        expect(response).to have_http_status(200)
+        expect(response).to render_template(:index)
+      end
+      it 'returns a 200 status code and render template for show' do
+        get '/users/1/posts', params: { id: 1 }
+        expect(response.body).to include('post')
+      end
+    end
+  end
```

Comment:
- [ ] Only 1 out of 10 of your users and posts controller tests passed. Kindly fix the remaining tests.

  ![spec-new](https://user-images.githubusercontent.com/54848363/181300128-4f3740e5-cf5c-4a56-b3d2-00d579570805.jpg)